Gems/Atom/RPI/Code/Source/RPI.Public/RenderPipeline.cpp

```diff
@@ -948,5 +950,93 @@ namespace AZ
+            if (aaMethod == AntiAliasingMode::Default)
+            {
+                return false;
+            }
+            m_activeAAMethod = aaMethod;
+            return SetAAMethod(this, m_activeAAMethod);
+        }
+
+        AntiAliasingMode RenderPipeline::GetActiveAAMethod()
+        {
+            return m_activeAAMethod;
+        }
+
+        AntiAliasingMode RenderPipeline::GetAAMethodByName(AZStd::string aaMethodName)
+        {
+            static AZStd::unordered_map<AZStd::string, AntiAliasingMode> AAMethodsLookup = {
+                {"MSAA", AntiAliasingMode::MSAA}, {"SMAA", AntiAliasingMode::SMAA}, 
+                {"TAA", AntiAliasingMode::TAA}
+            };
+ 
+            auto findIt = AAMethodsLookup.find(aaMethodName);
+            if (findIt != AAMethodsLookup.end())
+            {
+                return findIt->second;
+            }
+            return AntiAliasingMode::Default;
+        }        
+
+        AZStd::string RenderPipeline::GetAAMethodNameByIndex(AntiAliasingMode aaMethodIndex)
+        {
+            static AZStd::unordered_map<AntiAliasingMode, AZStd::string> AAMethodNameLookup = {
+                {AntiAliasingMode::MSAA, "MSAA"}, {AntiAliasingMode::SMAA, "SMAA"}, 
+                {AntiAliasingMode::TAA, "TAA"}
+            };
+ 
+            auto findIt = AAMethodNameLookup.find(aaMethodIndex);
+            if (findIt != AAMethodNameLookup.end())
+            {
+                return findIt->second;
+            }
+            return "MSAA";
+        }
+
+        bool RenderPipeline::EnablePass(RenderPipeline* pipeline, Name& passName, bool enable)
+        {
+            PassFilter passFilter = PassFilter::CreateWithPassName(passName, pipeline);
+            Ptr<Pass> aaPass = PassSystemInterface::Get()->FindFirstPass(passFilter);
+            if (!aaPass)
+            {
+                return false;
+            }
+            if (aaPass->IsEnabled() != enable)
+            {
+                aaPass->SetEnabled(enable);
+            }
+            return true;
+        }
+
+        bool RenderPipeline::SetAAMethod(RenderPipeline* pipeline, AZStd::string aaMethodName)
+        {
+            AntiAliasingMode aaMethod = GetAAMethodByName(aaMethodName);
+            return SetAAMethod(pipeline, aaMethod);
+        }
+
+        bool RenderPipeline::SetAAMethod(RenderPipeline* pipeline, AntiAliasingMode aaMethod)
+        {
+            if (aaMethod == AntiAliasingMode::Default)
+            {
+                return false;
+            }
+
+            static AZStd::unordered_map<AntiAliasingMode, Name> AAPassNamesLookup = {
+                {AntiAliasingMode::SMAA, Name("SMAA1xApplyLinearHDRColorPass")}, 
+                {AntiAliasingMode::TAA, Name("TaaPass")}
+            };
+
+            for (auto& aaPassMap : AAPassNamesLookup)
+            {
+                EnablePass(pipeline, aaPassMap.second, aaPassMap.first == aaMethod);
+            }
```

Comment:
Don't we also need to enable the `ContrastAdaptiveSharpeningPass` when TAA is enabled? and disable it when not?